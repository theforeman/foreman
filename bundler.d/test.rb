group :test do
  gem 'mocha', '~> 1.1', :require => false
  gem 'simplecov', '~> 0.9'
  gem 'spork-minitest', '0.0.3'
  gem 'single_test', '~> 0.6'
  gem 'minitest', '5.1.0'
  gem 'minitest-spec-rails' #, '~> 4.7'
  #gem 'ci_reporter', '2.0.0', :require => false
  gem 'ci_reporter_minitest', :require => false
  gem 'capybara', '~> 2.0'
  gem 'database_cleaner', '~> 1.3'
  gem 'launchy', '~> 2.4'
  gem 'spork', '~> 0.9'
  gem 'factory_girl_rails', '~> 4.5', :require => false
  gem 'rubocop-checkstyle_formatter', '~> 0.1'
  gem "poltergeist"
end
