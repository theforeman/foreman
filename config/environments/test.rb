# Track deprecation warnings in test environment as early as possible, but pause processing of
# deprecations until all plugins are registered (prior to the finisher_hook initializer) to ensure
# the whitelist is fully configured. This is done in the after_initialize block below.
ASDeprecationTracker.pause!

Foreman::Application.configure do
  # Settings specified here will take precedence over those in config/application.rb.

  # The test environment is used exclusively to run your application's
  # test suite. You never need to work with it otherwise. Remember that
  # your test database is "scratch space" for the test suite and is wiped
  # and recreated between test runs. Don't rely on the data there!
  config.cache_classes = true

  # Do not eager load code on boot. This avoids loading your whole application
  # just for the purpose of running a single test. If you are using a tool that
  # preloads Rails for running tests, you may have to set it to true.
  config.eager_load = false

  # Configure static asset server for tests with Cache-Control for performance.
  config.serve_static_files   = true
  config.static_cache_control = 'public, max-age=3600'

  # Adds additional error checking when serving assets at runtime.
  # Checks for improperly declared sprockets dependencies.
  # Raises helpful error messages.
  config.assets.raise_runtime_errors = true

  # Show full error reports and disable caching.
  config.consider_all_requests_local       = true
  config.action_controller.perform_caching = false

  # Raise exceptions instead of rendering exception templates.
  config.action_dispatch.show_exceptions = false

  # Disable request forgery protection in test environment.
  config.action_controller.allow_forgery_protection = false

  # Tell Action Mailer not to deliver emails to the real world.
  # The :test delivery method accumulates sent emails in the
  # ActionMailer::Base.deliveries array.
  config.action_mailer.delivery_method = :test
  config.action_mailer.perform_deliveries = true
  ActionMailer::Base.deliveries.clear

  # Use SQL instead of Active Record's schema dumper when creating the test database.
  # This is necessary if your schema can't be completely dumped by the schema dumper,
  # like if you have constraints or database-specific column types
  # config.active_record.schema_format = :sql

  # Should ANSI color codes be used when logging information
  config.colorize_logging = Foreman::Logging.config[:colorize]

  # Print deprecation notices to the stderr.
  config.active_support.deprecation = :stderr

  # Raise exception on mass assignment of unfiltered parameters
  config.action_controller.action_on_unpermitted_parameters = :strict

  # Use separate cache stores for parallel_tests
  config.cache_store = :file_store, Rails.root.join("tmp", "cache", "paralleltests#{ENV['TEST_ENV_NUMBER']}")

  # Enable automatic creation/migration of the test DB when running tests
  config.active_record.maintain_test_schema = true

  # Randomize the order test cases are executed.
  config.active_support.test_order = :random

  config.webpack.dev_server.enabled = false

  # Whitelist all plugin engines by default from raising errors on deprecation warnings for
  # compatibility, allow them to override it by adding an ASDT configuration file.
  config.after_initialize do
    Foreman::Plugin.all.each do |plugin|
      unless File.exist?(File.join(plugin.path, 'config', 'as_deprecation_whitelist.yaml'))
        ASDeprecationTracker.whitelist.add(engine: plugin.id.to_s.gsub('-', '_'))
      end
    end
    ASDeprecationTracker.resume!
  end
end
