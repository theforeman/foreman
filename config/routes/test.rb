if Rails.env.test?
  Foreman::Application.routes.draw do
    resources :testable, :only => :index
    resources :testable_strong_params, :only => :create
    resources :my_hosts, :only => :create
    resources :testable_no_params, :only => :create

    namespace :api do
      namespace :v2 do
        resources :testable, :only => [:create, :index]
      end

      resources :testable, :only => :index do
        get :raise_error, :on => :collection
      end
    end
  end
end
