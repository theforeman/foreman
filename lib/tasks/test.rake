namespace :test do
  desc "Test API"
  Rake::TestTask.new(:api) do |t|
    t.libs << "test"
    t.pattern = 'test/functional/api/**/*_test.rb'
    t.verbose = true
  end
end

namespace :test do
  desc "Test lib source"
  Rake::TestTask.new(:lib) do |t|
    t.libs << "test"
    t.pattern = 'test/lib/**/*_test.rb'
    t.verbose = true
  end
end

Rake::Task[:test].enhance do
  Rake::Task['test:lib'].invoke
end
