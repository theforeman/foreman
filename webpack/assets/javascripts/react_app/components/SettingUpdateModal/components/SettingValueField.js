import React from 'react';
import PropTypes from 'prop-types';
import { Col, HelpBlock, FormGroup, FormControl } from 'patternfly-react';
import classNames from 'classnames';

import { translate as __ } from '../../../common/I18n';
import { arraySelection } from '../../SettingsTable/SettingsTableHelpers';
import { renderOptions } from '../../common/forms/SelectHelpers';

const SettingValueField = ({ setting, form, field }) => {
  const { selectValues } = setting;
  const cssClasses = classNames({ 'masked-input': setting.encrypted });

  let inputField = <FormControl {...field} className={cssClasses} />;

  const error = form.errors && form.errors.value;

  if (selectValues) {
    inputField = (
      <FormControl {...field} componentClass="select" className={cssClasses}>
        {renderOptions(arraySelection(setting) || selectValues)}
      </FormControl>
    );
  }

  if (setting.settingsType === 'boolean') {
    inputField = (
      <FormControl {...field} componentClass="select" className={cssClasses}>
        <option value>{__('Yes')}</option>
        <option value={false}>{__('No')}</option>
      </FormControl>
    );
  }

  if (setting.settingsType === 'array') {
    inputField = (
      <FormControl
        {...field}
        componentClass="textarea"
        className={cssClasses}
      />
    );
  }

  const helpBlock = (
    <HelpBlock>
      <span className="error-msg">{error}</span>
    </HelpBlock>
  );

  const encryptedHelp = (
    <HelpBlock>
      {__(
        'This setting is encrypted. Empty input field is displayed instead of the setting value.'
      )}
    </HelpBlock>
  );

  return (
    <React.Fragment>
      <FormGroup className={error ? 'has-error' : ''}>
        <Col md={10}>
          {inputField}
          {setting.encrypted && encryptedHelp}
        </Col>
        {error && helpBlock}
      </FormGroup>
    </React.Fragment>
  );
};

SettingValueField.propTypes = {
  setting: PropTypes.object.isRequired,
  form: PropTypes.object.isRequired,
  field: PropTypes.object.isRequired,
};

export default SettingValueField;
