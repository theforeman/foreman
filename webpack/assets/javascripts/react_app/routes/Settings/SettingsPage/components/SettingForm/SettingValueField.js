import React from 'react';
import PropTypes from 'prop-types';

import { Col, HelpBlock, FormGroup } from 'patternfly-react';

import InputFactory from '../../../../../components/common/forms/InputFactory';

const SettingValueField = ({ setting, form, field }) => {
  const error = form.errors && form.errors.value;

  const arraySelectProps = settingModel =>
    settingModel.selectValues && settingModel.selectValues.kind === 'array'
      ? { type: 'arraySelect', field, model: settingModel }
      : null;

  const hashSelectProps = settingModel =>
    settingModel.selectValues && settingModel.selectValues.kind === 'hash'
      ? { type: 'hashSelect', field, model: settingModel }
      : null;

  const boolSelectProps = settingModel =>
    settingModel.settingsType === 'boolean'
      ? { type: 'boolSelect', field }
      : null;

  const arrayProps = settingModel =>
    settingModel.settingsType === 'array'
      ? { ...field, componentClass: 'textarea' }
      : null;

  const inputProps = settingModel => ({ ...field });

  const fieldProps = (propSelectors, model) =>
    propSelectors.reduce((memo, propFn) => {
      if (memo) {
        return memo;
      }

      return propFn(model);
    }, null);

  const factoryProps = fieldProps(
    [
      arraySelectProps,
      hashSelectProps,
      boolSelectProps,
      arrayProps,
      inputProps,
    ],
    setting
  );

  const helpBlock = (
    <HelpBlock>
      <span className="error-msg">{error}</span>
    </HelpBlock>
  );

  return (
    <React.Fragment>
      <FormGroup className={error ? 'has-error' : ''}>
        <Col md={10}>
          <InputFactory {...factoryProps} />
        </Col>
        {error && helpBlock}
      </FormGroup>
    </React.Fragment>
  );
};

SettingValueField.propTypes = {
  setting: PropTypes.object.isRequired,
  form: PropTypes.object.isRequired,
  field: PropTypes.object.isRequired,
};

export default SettingValueField;
