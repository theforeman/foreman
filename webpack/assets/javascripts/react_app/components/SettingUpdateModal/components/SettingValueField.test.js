import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import SettingValueField from './SettingValueField';

import {
  arraySetting,
  withArraySelection,
  withHashSelection,
  boolSetting,
  stringSetting,
  rootPass,
} from '../../SettingRecords/__tests__/SettingRecords.fixtures';

const baseFixtures = {
  form: {},
  field: {},
};

const fixtures = {
  'should render for string setting': {
    setting: stringSetting,
    ...baseFixtures,
  },
  'should render for array setting': {
    setting: arraySetting,
    ...baseFixtures,
  },
  'should render for bool setting': {
    setting: boolSetting,
    ...baseFixtures,
  },
  'should render for setting with hash selection': {
    setting: withHashSelection,
    ...baseFixtures,
  },
  'should render for setting with array selection': {
    setting: withArraySelection,
    ...baseFixtures,
  },
  'should render for encrypted setting': {
    setting: rootPass,
    ...baseFixtures,
  },
};

describe('SettingValueField', () =>
  testComponentSnapshotsWithFixtures(SettingValueField, fixtures));
