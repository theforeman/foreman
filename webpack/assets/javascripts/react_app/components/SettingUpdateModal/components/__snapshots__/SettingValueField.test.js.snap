// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingValueField should render for array setting 1`] = `
<Fragment>
  <FormGroup
    bsClass="form-group"
    className=""
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={10}
    >
      <FormControl
        bsClass="form-control"
        className=""
        componentClass="textarea"
      />
    </Col>
  </FormGroup>
</Fragment>
`;

exports[`SettingValueField should render for bool setting 1`] = `
<Fragment>
  <FormGroup
    bsClass="form-group"
    className=""
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={10}
    >
      <FormControl
        bsClass="form-control"
        className=""
        componentClass="select"
      >
        <option
          value={true}
        >
          Yes
        </option>
        <option
          value={false}
        >
          No
        </option>
      </FormControl>
    </Col>
  </FormGroup>
</Fragment>
`;

exports[`SettingValueField should render for encrypted setting 1`] = `
<Fragment>
  <FormGroup
    bsClass="form-group"
    className=""
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={10}
    >
      <FormControl
        bsClass="form-control"
        className="masked-input"
        componentClass="input"
      />
      <HelpBlock
        bsClass="help-block"
      >
        This setting is encrypted. Empty input field is displayed instead of the setting value.
      </HelpBlock>
    </Col>
  </FormGroup>
</Fragment>
`;

exports[`SettingValueField should render for setting with array selection 1`] = `
<Fragment>
  <FormGroup
    bsClass="form-group"
    className=""
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={10}
    >
      <FormControl
        bsClass="form-control"
        className=""
        componentClass="select"
      >
        <option
          key="null"
          value=""
        >
          Select an owner
        </option>
        <optgroup
          key="Users"
          label="Users"
        >
          <option
            key="13-Users"
            value="13-Users"
          >
            canned_admin
          </option>
          <option
            key="1"
            value="19-Users"
          >
            user
          </option>
          <option
            key="2"
            value="27-Users"
          >
            viewer
          </option>
          <option
            key="3"
            value="4-Users"
          >
            admin
          </option>
        </optgroup>
        <optgroup
          key="Usergroups"
          label="Usergroups"
        >
          <option
            key="1-Usergroups"
            value="1-Usergroups"
          >
            basic broup
          </option>
          <option
            key="1"
            value="2-Usergroups"
          >
            view hosts group
          </option>
        </optgroup>
      </FormControl>
    </Col>
  </FormGroup>
</Fragment>
`;

exports[`SettingValueField should render for setting with hash selection 1`] = `
<Fragment>
  <FormGroup
    bsClass="form-group"
    className=""
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={10}
    >
      <FormControl
        bsClass="form-control"
        className=""
        componentClass="select"
      >
        <option
          key="CoreOS PXELinux"
          value="CoreOS PXELinux"
        >
          CoreOS PXELinux
        </option>
        <option
          key="FreeBSD (mfsBSD) PXELinux"
          value="FreeBSD (mfsBSD) PXELinux"
        >
          FreeBSD (mfsBSD) PXELinux
        </option>
        <option
          key="Kickstart default PXELinux"
          value="Kickstart default PXELinux"
        >
          Kickstart default PXELinux
        </option>
        <option
          key="Kickstart oVirt-RHVH PXELinux"
          value="Kickstart oVirt-RHVH PXELinux"
        >
          Kickstart oVirt-RHVH PXELinux
        </option>
        <option
          key="Preseed default PXELinux"
          value="Preseed default PXELinux"
        >
          Preseed default PXELinux
        </option>
        <option
          key="PXELinux chain iPXE"
          value="PXELinux chain iPXE"
        >
          PXELinux chain iPXE
        </option>
        <option
          key="PXELinux chain iPXE UNDI"
          value="PXELinux chain iPXE UNDI"
        >
          PXELinux chain iPXE UNDI
        </option>
        <option
          key="PXELinux default local boot"
          value="PXELinux default local boot"
        >
          PXELinux default local boot
        </option>
        <option
          key="PXELinux default memdisk"
          value="PXELinux default memdisk"
        >
          PXELinux default memdisk
        </option>
        <option
          key="PXELinux global default"
          value="PXELinux global default"
        >
          PXELinux global default
        </option>
        <option
          key="RancherOS PXELinux"
          value="RancherOS PXELinux"
        >
          RancherOS PXELinux
        </option>
        <option
          key="[templates] A fake pxelinux"
          value="[templates] A fake pxelinux"
        >
          [templates] A fake pxelinux
        </option>
        <option
          key="TEST default"
          value="TEST default"
        >
          TEST default
        </option>
        <option
          key="WAIK default PXELinux"
          value="WAIK default PXELinux"
        >
          WAIK default PXELinux
        </option>
        <option
          key="Windows default PXELinux"
          value="Windows default PXELinux"
        >
          Windows default PXELinux
        </option>
        <option
          key="XenServer default PXELinux"
          value="XenServer default PXELinux"
        >
          XenServer default PXELinux
        </option>
      </FormControl>
    </Col>
  </FormGroup>
</Fragment>
`;

exports[`SettingValueField should render for string setting 1`] = `
<Fragment>
  <FormGroup
    bsClass="form-group"
    className=""
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={10}
    >
      <FormControl
        bsClass="form-control"
        className=""
        componentClass="input"
      />
    </Col>
  </FormGroup>
</Fragment>
`;
